nncf/common/composite_compression.py

```diff
@@ -216,26 +216,24 @@ def compression_stage(self) -> CompressionStage:
                 result += current_level
         return result
 
-    def load_state(self, states: List[Dict[str, object]]) -> None:
-        """
-        Calls `load_state()` method for all children.
-
-        :param states: Output of `get_state()` method.
-        """
-        for child_ctrl, child_state in zip(self.child_ctrls, states):
-            child_ctrl.load_state(child_state)
-
-    def get_state(self) -> List[Dict[str, object]]:
+    def get_state(self) -> Dict[str, Dict]:
         """
         Returns the composite compression controller state. This state contains
         the state of all children.
-
         :return: The composite compression controller state.
         """
-        composite_state = []
-        for child_ctrl in self.child_ctrls:
-            composite_state.append(child_ctrl.get_state())
-        return composite_state
+        result = dict()
+        for ctrl in self.child_ctrls:
+            result.update(ctrl.get_state())
+        return result
+
+    def load_state(self, state: Dict[str, Dict]) -> None:
+        """
+        Loads the compression controller state of children
+        :param state: Output of `get_state()` method.
+        """
+        for ctrl in self.child_ctrls:
+            ctrl.load_state(state)
```

Comment:
The `state` argument of the `CompressionAlgorithmController.load_state()` method is an output of the `CompressionAlgorithmController.get_state()` method. But here you use the output of the `CompositeCompressionAlgorithmController.get_state()` method. I think you need to write something like this

```python
for ctrl in self.child_ctrls:
    ctrl_state = {
        ctrl.name: state.get(ctrl.name, {})
    }
    ctrl.load_state(ctrl_state)
```

Reply:
No, it works fine for child controllers, because they find state by name.
It's tested in `test_ctrl_state.py`
```python
    def load_state(self, state: Dict[str, Any]) -> None:
        if self.name in state:
            algo_state = state[self.name]
            self.loss.load_state(algo_state[self._state_names.LOSS])
            self.scheduler.load_state(algo_state[self._state_names.SCHEDULER])
```

Reply:
It works but it conflicts with the docstring of the `CompressionAlgorithmController.load_state()` method.
@alexsu52 @vshampor please resolve this conversation if it looks good for you

Reply:
If to be picky, your proposal above would conflict with docstring as well. because it's not output of `get_state()`.
I think confusion can be removed by rewriting docstring for `load_state` and `get_state` in the following way:
```
    def get_state(self) -> Dict[str, Any]:
        """
        Returns compression controller state, which is the map of the algorithm name to the dictionary with the corresponding state attributes.
        """

    def load_state(self, state: Dict[str, Any]) -> None:
        """
        Loads the compression controller state from the map of algorithm name to the dictionary with state attributes

        :param: map of the algorithm name to the dictionary with the corresponding state attributes.
        """
```

Reply:
Corrected docstring and typehints